Problems with installing OpenBLAS: error relocation R_X86_64_32S against undefined symbol `PREFETCHSIZE'

I am trying to install OpenBLAS on a new laptop (Processor model Intel(R) Core(TM) Ultra 7 258V, 8 core) on Ubuntu-24.04 running under WSL.  The application that uses OpenBLAS needs 64-bit integer support, so I am building from source.  The following command has worked for me on older processors:
make BINARY=64 INTERFACE64=1 PREFIX=<my home directory>

Issue 1: This form of the make command produced 10 instances of the following error message with references to different lines in generic/dot.c :

/usr/lib/gcc/x86_64-linux-gnu/13/include/fmaintrin.h:63:1: error: inlining failed in call to ‘always_inline’ ‘_mm256_fmadd_ps’: target specific option mismatch
   63 | _mm256_fmadd_ps (__m256 __A, __m256 __B, __m256 __C)
      | ^~~~~~~~~~~~~~~
In file included from ../kernel/x86_64/../generic/../simd/intrin.h:63:
../kernel/x86_64/../generic/../simd/intrin_avx.h:22:26: note: called from here
   22 |     #define v_muladd_f32 _mm256_fmadd_ps
../kernel/x86_64/../generic/dot.c:69:21: note: in expansion of macro ‘v_muladd_f32’
   69 |             vsum3 = v_muladd_f32(

Following various suggestions on the web, I was able to modify the make command to the following and got past these messages:
make BINARY=64 INTERFACE64=1 PREFIX=<my home directory> CFLAGS+='-march=native -mavx'
However, I am not sure why I would need to do this and whether it really fixed the problem.


Issue 2: with the modified make command, the build process get further, but dies during the library build with undefined symbol 'PREFETCHSIZE'

Here is the message:
/usr/bin/ld: ../libopenblas_prescottp-r0.3.30.a(ssymv_U.o): relocation R_X86_64_32S against undefined symbol `PREFETCHSIZE' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: failed to set dynamic section sizes: bad value
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:207: ../libopenblas_prescottp-r0.3.30.so] Error 1
make[1]: Leaving directory '/home/swope/OpenBLAS/OpenBLAS-0.3.30/exports'
make: *** [Makefile:149: shared] Error 2

I followed the suggestion to recompile with -fPIC by adding it to CFLAGS+='-march=native -mavx -fPIC'
However, this did not fix the problem.  It died at exactly the same place with the same message.

I am kind of stuck and any suggestions would be appreciated:
1) is my processor model supported by OpenBLAS?
2) was addition of CFLAGS='-march=native -mavx' the correct way to fix issue #1?  Why was this needed when it wasn't needed before?
3) how do I fix issue #2 with the undefined PREFETCHSIZE.

Thank you in advance for any help or suggestions.....Bill

--- a/cpuid_x86.c
+++ b/cpuid_x86.c
@@ -1567,6 +1567,7 @@ int get_cpuname(void){
           case 10:
           case 15:
 	  case 14: // Alder Lake N
+	  case 13: // Lunar Lake
             if(support_avx2())
               return CPUTYPE_HASWELL;
 	    if(support_avx())
@@ -2412,7 +2413,8 @@ int get_coretype(void){
 	  case 7: // Raptor Lake
           case 10:
           case 15:
-	  case 14: // Alder Lake N	  
+	  case 14: // Alder Lake N
+	  case 13: // Lunar Lake
 #ifndef NO_AVX2
 	  if(support_avx2())
             return CORE_HASWELL;

